satip/download.py

```diff
@@ -0,0 +1,303 @@
+                format_dt_str(end_time),
+                product_id=product_id,
+            )
+        # Sanity check, able to open/right size and move to correct directory
+        sanity_check_files_and_move_to_directory(
+            directory=download_directory, product_id=product_id
+        )
+    pass
+
+
+def load_key_secret(filename) -> Tuple[str, str]:
+    """
+    Load user secret and key stored in a yaml file
+
+    Args:
+        filename: Filename to read
+
+    Returns:
+        The user secret and key
+
+    """
+    with fsspec.open(filename, mode="r") as f:
+        keys = yaml.load(f, Loader=yaml.FullLoader)
+        return keys["key"], keys["secret"]
+
+
+def sanity_check_files_and_move_to_directory(directory, product_id):
+    """
+    Runs a sanity check for all the files of a given product_id in the directory
+    Deletes incomplete files, moves checked files to final location
+
+    This does a sanity check by:
+        Checking the filesize of RSS images
+        Attempting to open them in SatPy
+
+    While loading the file should be somewhat slow, it ensures the file can be opened
+
+    Args:
+        directory: Directory where the native files were downloaded
+        product_id: The product ID of the files to check
+
+    Returns:
+        The number of incomplete files deleted
+    """
+    pattern = "*.nat" if product_id == RSS_ID else "*.grb"
+    fs = fsspec.filesystem("file")  # TODO Update for other ones?
+    new_files = fs.glob(os.path.join(directory, pattern))
+
+    date_func = (
+        eumetsat_native_filename_to_datetime
+        if product_id == RSS_ID
+        else eumetsat_cloud_name_to_datetime
+    )
+    # Check all the right size
+    satpy_reader = "seviri_l1b_native" if product_id == RSS_ID else "seviri_l2_grib"
+
+    if product_id == RSS_ID:
+        for f in new_files:
+            try:
+                file_size = eumetsat.get_filesize_megabytes(f)
+                if not math.isclose(file_size, NATIVE_FILESIZE_MB, abs_tol=1):
+                    print("Wrong Size")
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore", UserWarning)
+                    scene = Scene(filenames=[f], reader=satpy_reader)
+                    scene.load(SAT_VARIABLE_NAMES)
+                # Now that the file has been checked and can be open, move it to the final directory
+                base_name = get_basename(f)
+                file_date = date_func(base_name)
+                # Want to move it 1 minute in the future to correct the difference
+                file_date = file_date + timedelta(minutes=1)
+                if not fs.exists(os.path.join(directory, file_date.strftime(format="%Y/%m/%d"))):
+                    fs.mkdir(os.path.join(directory, file_date.strftime(format="%Y/%m/%d")))
+                fs.move(
+                    f, os.path.join(directory, file_date.strftime(format="%Y/%m/%d"), base_name)
+                )
+            except:
```

Comment:
We should probably log the exception, just in case it's something unexpected.

```suggestion
            except:
                _LOG.exception(f"Error when sanity-checking {f}.  Deleting this file.  Will be downloaded next time this script is run.")
```

(IIRC, `LOG.exception()` will automatically log the traceback.)